Update authorization-related docs for billing release (#2212)
clerk/clerk-docs: docs/components/protect.mdx

---
title: '`<Protect>`'

> description: The Protect component is used for authorization checks. It only renders its children when the current user has the specified Permission or Role in the Organization.

Reviewer: ```suggestion
description: The Protect component is used for authorization checks. It only renders its children when the current user has the specified access.
```

Since its based on the user having the correct permission, role, plan or feature now. May be good to just say its once they have access.

  - `fallback?`
  - `JSX`

  Optional UI to show when a user doesn't have the correct type of access control to access the protected content.

  ---

  - `feature?`
  - `string`

  Optional string corresponding to a [feature](/docs/billing/overview).

  ---

  - `plan?`
  - `string`

  Optional string corresponding to a [plan](/docs/billing/overview).

  ---

  - `permission?`
  - `string`

  Optional string corresponding to a [permission](/docs/organizations/roles-permissions) in the format `org:<resource>:<action>`

  ---

  - `role?`
  - `string`

  Optional string corresponding to a [role](/docs/organizations/roles-permissions) in the format `org:<role>`

  ---

  - `treatPendingAsSignedOut?`
  - `boolean`

  A boolean that indicates whether to treat pending sessions as signed out. Defaults to `false`.
</Properties>

## Usage

> To limit who is able to see the content that `<Protect>` renders, you can pass **one** of the access control props: `permission`, `role`, `feature`, or `plan`. It's recommended to use permission-based authorization over role-based authorization, as it's more flexible, easier to manage, and more secure. Permissions can be assigned to different roles with ease.

Reviewer: This could be a little confusing, if a developer is aiming to use subscription based access control using the `plan` and `feature` props, but this is telling you it's recommended to use permission-based checks, could send a developer down the wrong way, trying to bind feature access to permissions or something funky.

        <template #fallback>
          <p>Only a member of the Billing department can access this content.</p>
        </template>
        <p>Users with role org:billing can see this.</p>
      </Protect>
    </template>
    ```
  </Tab>
</Tabs>

### Render content by plan

> The following example demonstrates how to use `<Protect />` to protect content by checking if the user has a plan.

Reviewer 2: @alexisintech I think we also wanna mention something among the lies of  "While authorization by `feature` is **recommended**, for convenience, `<Protect>` allows a `plan` prop to be passed."

That is because, you as a developer can change plans but use the same features. That will not cause your app to break. Similar to how we recommend permissions over roles.

Author: great suggestion !
